### ISSUE TITLE
How to set Meteor.Settings

### ISSUE THREAD

**Issue by emgee3** [emgee3/meteor-mssql#10]
(Copied from a different, unrelated issue.)

@yanxiaoyan:

```
Hi there,
I am a learner,
【meteor.settings】Should be how to set it?
```

@batjko:

```
You create a settings.json in your project folder, with the settings object inside. 
Then you start meteor with this parameter: meteor --settings settings.json

Then the settings will be available to your Meteor app as Meteor.settings, and this package automatically uses it to establish a database connection.
```

@yanxiaoyan:

```
I created a settings.json in my project,but can not connect to the database,I think it should be the configuration parameter is not right, server this add port number?
```

**Comment by emgee3** [comment]
@yanxiaoyan

Your [`settings.json`](https://github.com/emgee3/meteor-mssql#settings) file does not need to be in the project, but you do need to reference this file when launching Meteor. Example: 

```
$ meteor --settings /path/to/settings.json
```

You can see the format for the database settings object here: https://github.com/patriksimek/node-mssql#configuration-1

**Comment by WangYans** [comment]
@emgee3 How can I get to the data? API is not very clear to me. I hope to give me a look,If you have a small case, I hope you can send me a copy. very grateful.

**Comment by emgee3** [comment]
The `Sql` object is only exposed on the server. So if you have the following code on the server: 

```
if (Meteor.isServer) {
    Meteor.startup(function () {
        Sql.q("select 1 as num", function (err, res) {
            console.log(res);
        });
    });
}
```

You should see something like:

```
I20150908-21:30:48.708(-7)? { num: 1 }
```

logged to the console.

Note: I didn't test this code, you might need to put a delay for the driver to connect to the database before you can run a query.

**Comment by WangYans** [comment]
@emgee3 I still don't understand, this can only be written in the server?

**Comment by emgee3** [comment]
Correct, the `Sql` object is server-only. You do not want clients able to write arbitrary SQL code. So if you need to get the results of a query to the client, wrap it in a Meteor Method, or add code to insert into a collection. See http://docs.meteor.com/#/full/meteor_methods and http://docs.meteor.com/#/full/meteor_publish (relevant part starts with _"Alternatively, a publish function can directly control its published record set by calling the..."_)

**Comment by WangYans** [comment]
@emgee3 I'm not quite understand,how should I add code to insert into a collection

**Comment by yanxiaoyan** [comment]
<template name="fengongsiList">
  <div class="posts" style="border:2px solid blue">
    {{#each fengongsi}}
      {{> newTitle}}
    {{/each}}
  </div>
</template>

**Comment by yanxiaoyan** [comment]
<template name="fengongsiList">
  <div class="posts" style="border:2px solid blue">
    {{#each fengongsi}}
      {{> newTitle}}
    {{/each}}
  </div>
</template>

**Comment by yanxiaoyan** [comment]
if (Meteor.isServer) {
    Meteor.startup(function () {
        Sql.q("select 1 as num", function (err, res) {
            console.log(res);
        });
    });
}

**Comment by yanxiaoyan** [comment]
Template.newsList.helpers({
	fengongsi:function(){
		var opts = {
	    	query : "select * from fengongsi", 
		}

		try {
		    var query = Sql.ps(opts);
		} catch (e) {
		     
		}
		var result  = query();

		return result;
	}
});

**Comment by yanxiaoyan** [comment]
Template.newsList.helpers({
	fengongsi:function(){
		var opts = {
	    	query : "select * from fengongsi", 
		}
		try {
		    var query = Sql.ps(opts);
		} catch (e) {     
		}
		var result  = query();
		return result;
	}
});

**Comment by yanxiaoyan** [comment]
Template.newsList.helpers({

	fengongsi:function(){
		var opts = {

	    	     query : "select * from fengongsi", 
		}
		try {
		    var query = Sql.ps(opts);
		} catch (e) {     
		}

		var result  = query();

		return result;
	}
});

**Comment by yanxiaoyan** [comment]
Template.newsList.helpers({
	fengongsi:function(){
		var opts = {
	    	    query : "select * from fengongsi", 
		}
		try {
		    var query = Sql.ps(opts);
		} catch (e) {     
		}
		var result  = query();
		return result;
	}
 });

**Comment by yanxiaoyan** [comment]
![image](https://cloud.githubusercontent.com/assets/14175238/9756964/5a025794-5710-11e5-99d1-11a698a49cf7.png)

**Comment by yanxiaoyan** [comment]
@emgee3 I put it in the method, but there is still a mistake, how to change this?

**Comment by WangYans** [comment]
@emgee3 I put it in the method, but there is still a mistake, how to change this? very grateful.

![image](https://cloud.githubusercontent.com/assets/14175238/9757052/c4fb616c-5710-11e5-8089-702f8ec2ac1e.png)
![image](https://cloud.githubusercontent.com/assets/14175238/9757060/cac57f42-5710-11e5-93fc-b26006d08e25.png)
![image](https://cloud.githubusercontent.com/assets/14175238/9757154/4993b532-5711-11e5-9948-9aa9873ed96b.png)
![image](https://cloud.githubusercontent.com/assets/14175238/9757158/515d62c2-5711-11e5-80c2-3ab762361ccf.png)

**Comment by emgee3** [comment]
So what you're running into isn't really a `emgee:mssql` problem, there's a bit more you have to learn about how async works in Meteor. Your template helper isn't going to wait until the async Meteor.call completes, so your query results aren't going to be available to the template. There are different strategies to handle this, and I can't really teach a Meteor class in this issue.

What I suggest is [Discover Meteor](http://discovermeteor.com), the chapter on Advanced Reactivity goes over this general subject.

A couple other hints in your code -- you should prepare the SQL statement outside the method. You only need to prepare the statement once, then you can use multiple times. Also, your try-catch should contain the query itself.
